### PR TITLE
legacy: Remove CoreOS images

### PR DESCRIPTION
**Pull request by tfussell**
In Azure and KVM, we decided not to support CoreOS at all in upcoming operator releases so I'm removing the CoreOS AMIs here to match.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/key/generate/generate_amis.go [hunk 1]
```diff
@@ -104,45 +102,32 @@ func scrapeVersionAMIs(source io.Reader) (map[string]string, error) {
 }
 
 func main() {
```
Comment (MarcelMue) [inline service/controller/key/generate/generate_amis.go:104]: Did we add some docs on how we do this? Something maybe in this operators readme would be good.

Comment (tfussell) [reply service/controller/key/generate/generate_amis.go:105]: Good call. I added some. Let me know what you think.

File: service/controller/key/amis.go [hunk 2]
```diff
@@ -1,249 +1,147 @@
+// NOTE: This file is generated. Do not edit.
```
Comment (MarcelMue) [inline service/controller/key/amis.go:1]: This seems to still be missing `china` images. Can this easily be extended to include them?

Comment (mazzy89) [reply service/controller/key/amis.go:1]: We generate as an artifact of the build to China a JSON file containing the `current` (latest) AMIs ID for China. It is public available

Comment (tfussell) [reply service/controller/key/amis.go:1]: Yep! I'll add it in a separate PR once I have the China JSON URL.

**Pull request merged**